src/effects/backends/audiounit/audiounitmanager.mm

```diff
@@ -0,0 +1,109 @@
+#import <AVFAudio/AVFAudio.h>
+#import <AudioToolbox/AudioToolbox.h>
+#include "util/assert.h"
+
+#include <QString>
+
+#include "effects/backends/audiounit/audiounitmanager.h"
+
+AudioUnitManager::AudioUnitManager(AVAudioUnitComponent* _Nullable component,
+        AudioUnitInstantiationType instantiationType)
+        : m_name(QString::fromNSString([component name])) {
+    // NOTE: The component can be null if the lookup failed in
+    // `AudioUnitBackend::createProcessor`, in which case the effect simply acts
+    // as an identity function on the audio. Same applies when
+    // `AudioUnitManager` is default-initialized.
+    if (!component) {
+        return;
+    }
+
+    switch (instantiationType) {
+    case Sync:
+        instantiateAudioUnitSync(component);
+        break;
+    case AsyncInProcess:
+    case AsyncOutOfProcess:
+        instantiateAudioUnitAsync(
+                component, instantiationType == AsyncInProcess);
+        break;
+    }
+}
+
+AudioUnitManager::~AudioUnitManager() {
+    if (m_isInstantiated.load()) {
+        qDebug() << "Uninitializing and disposing of Audio Unit" << m_name;
+        AudioUnitUninitialize(m_audioUnit);
+        AudioComponentInstanceDispose(m_audioUnit);
+    }
+}
+
+AudioUnit _Nullable AudioUnitManager::getAudioUnit() {
+    // We need to load this atomic flag to ensure that we don't get a partial
+    // read of the audio unit pointer (probably extremely uncommon, but not
+    // impossible: https://belkadan.com/blog/2023/10/Implicity-Atomic)
+    if (!m_isInstantiated.load()) {
```

Comment:
Are you using an atomic instead of a mutex because this method is called from the realtime audio thread? If so, a comment mentioning that would be good.

Reply:
That was what I assumed, `getAudioUnit` is called from `processChannel` which operates directly on sample buffers, though it might be worth checking whether that is indeed the audio thread. Since that's a method on the hot path, we want to avoid blocking there.

Reply:
Added a short note in 440d1cf4b4.